conf/modules.config

```diff
@@ -198,59 +198,99 @@ process {
-                path: { "${params.outdir}/multiqc" },
-                enabled: false
-            ]
+if (!params.skip_trimming) {
+    if (params.trimmer == 'trimgalore') {
+        process {
+            withName: '.*:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:TRIMGALORE' {
+                ext.args   = {
+                    [
+                        "--fastqc_args '-t ${task.cpus}'",
+                        params.extra_trimgalore_args ? params.extra_trimgalore_args.split("\\s(?=--)") : ''
+                    ].flatten().unique(false).join(' ').trim()
+                }
+                publishDir = [
+                    [
+                        path: { "${params.outdir}/${params.trimmer}/fastqc" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.{html,zip}"
+                    ],
+                    [
+                        path: { "${params.outdir}/${params.trimmer}" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.fq.gz",
+                        enabled: params.save_trimmed
+                    ],
+                    [
+                        path: { "${params.outdir}/${params.trimmer}" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.txt"
+                    ]
+                ]
+            }
+        }
+    }
+
+    if (params.trimmer == 'fastp') {
+        process {
+            withName: '.*:FASTQ_FASTQC_UMITOOLS_FASTP:FASTP' {
+                ext.args   = params.extra_fastp_args ?: ''
+                publishDir = [
+                    [
+                        path: { "${params.outdir}/${params.trimmer}" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.{json,html}"
+                    ],
+                    [
+                        path: { "${params.outdir}/${params.trimmer}/log" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.log"
+                    ],
+                    [
+                        path: { "${params.outdir}/${params.trimmer}" },
+                        mode: params.publish_dir_mode,
+                        pattern: "*.fastq.gz",
+                        enabled: params.save_trimmed
+                    ]
+                ]
+            }
         }
     }
 }
 
 if (params.with_umi && !params.skip_umi_extract) {
     process {
-        withName: '.*:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:UMITOOLS_EXTRACT' {
+        withName: 'UMITOOLS_EXTRACT' {
             ext.args   = [
                     params.umitools_extract_method ? "--extract-method=${params.umitools_extract_method}" : '',
                     params.umitools_bc_pattern     ? "--bc-pattern='${params.umitools_bc_pattern}'" : '',
                     params.umitools_bc_pattern2    ? "--bc-pattern2='${params.umitools_bc_pattern2}'" : ''
-                ].join(' ').trim()
+            ].join(' ').trim()
```

Comment:
I think we should add this still, since UMI-tools extract since version 1.14 also supports the separator parameter:
`params.umitools_umi_separator   ? "--umi-separator='${params.umitools_umi_separator}'" : '',`

Reply:
I can't see it as an option for this command?

```
root@6acbd26ea861:/home/harshil/repos/nf-core/viralrecon# umi_tools extract --help
UMI-Tools: Version 1.1.4
root@6acbd26ea861:/home/harshil/repos/nf-core/viralrecon# umi_tools extract 

extract - Extract UMI from fastq

Usage:

   Single-end:
      umi_tools extract [OPTIONS] -p PATTERN [-I IN_FASTQ[.gz]] [-S OUT_FASTQ[.gz]]

   Paired end:
      umi_tools extract [OPTIONS] -p PATTERN [-I IN_FASTQ[.gz]] [-S OUT_FASTQ[.gz]] --read2-in=IN2_FASTQ[.gz] --read2-out=OUT2_FASTQ[.gz]

   note: If -I/-S are ommited standard in and standard out are used
         for input and output.  To generate a valid BAM file on
         standard out, please redirect log with --log=LOGFILE or
         --log2stderr. Input/Output will be (de)compressed if a
         filename provided to -S/-I/--read2-in/read2-out ends in .gz
         

For full UMI-tools documentation, see https://umi-tools.readthedocs.io/en/latest/

Required options missing, see --help for more details
```

Reply:
We can add it in the next release.

Reply:
Yes, the docs have not been updated, but I tried it out yesterday and double-checked. 

You can also see it in the [release notes to version 1.13](https://github.com/CGATOxford/UMI-tools/releases/tag/1.1.3) and the corresponding [PR](https://github.com/CGATOxford/UMI-tools/pull/548).

Reply:
Added in https://github.com/nf-core/rnaseq/pull/977

Reply:
Thanks a lot! I think that is very useful, in particular since Illumina pushes for the `:` as the standard separator, so most tools will follow suit and struggle with the underscore of umi-tools deduplicated BAMs.